cdb2api/cdb2api.c

```diff
@@ -5081,9 +5115,53 @@ int cdb2_clearbindings(cdb2_hndl_tp *hndl)
     free(hndl->bindvars);
     hndl->bindvars = NULL;
     hndl->n_bindvars = 0;
+    free(hndl->bind_list_name);
+    hndl->bind_list_name = NULL;
+    free(hndl->bind_list_repl);
+    hndl->bind_list_repl = NULL;
     return 0;
 }
 
+/* bind @list with array of type/variable */
+int cdb2_bind_list(cdb2_hndl_tp *hndl, const char *varname, int count,
+                   int type[], const void *varaddr[], int length[])
+{
+    free(hndl->bind_list_name);
+    free(hndl->bind_list_repl);
+    int len = strlen(varname);
+    hndl->bind_list_name = malloc(len + 1);
+    hndl->bind_list_name[0] = '@';
+    strcpy(&hndl->bind_list_name[1], varname); // to search
+    hndl->bind_list_repl = malloc((len + 8) * count); // to replace
+    hndl->bind_list_repl[0] = '\0';
+
+    char loc_varname[len + 6];
```

Comment:
Why `+ 6`?  32-bit ints are up to 10 digits long, so for this to be correct it seems like you need to check if `0 <= count <= 9999` here.